fabric-command-api-v1/src/main/java/net/fabricmc/fabric/impl/command/v1/LegacyHandler.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.command.v1;
+
+import net.minecraft.server.command.CommandManager.RegistrationEnvironment;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
+
+public final class LegacyHandler implements ModInitializer {
+	@Override
+	@SuppressWarnings("deprecation")
+	public void onInitialize() {
+		CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback.EVENT.invoker().register(dispatcher, environment == RegistrationEnvironment.DEDICATED));
```

Comment:
Check `environment.dedicated` so that `RegistrationEnvironment.ALL` is handled properly.

Reply:
The implementation I am using matches the javadoc on v1 better and is the same as the existing implementation, although that can be a little misleading.